crypto/ui/ui_lib.c

```diff
@@ -43,6 +43,7 @@ UI *UI_new_method(const UI_METHOD *method)
     ret->meth = method;
 
     if (!CRYPTO_new_ex_data(CRYPTO_EX_INDEX_UI, ret, &ret->ex_data)) {
+        CRYPTO_THREAD_lock_free(ret->lock);
         OPENSSL_free(ret);
```

Comment:
It would probably be better to simply call `UI_free(ret)`...  I'm not entirely sure about the ramifications, though.